common/interactivetx.c

```diff
@@ -0,0 +1,687 @@
+#include "config.h"
+#include "interactivetx.h"
+#include <bitcoin/chainparams.h>
+#include <bitcoin/psbt.h>
+#include <bitcoin/script.h>
+#include <ccan/array_size/array_size.h>
+#include <ccan/cast/cast.h>
+#include <ccan/mem/mem.h>
+#include <ccan/tal/str/str.h>
+#include <common/billboard.h>
+#include <common/blockheight_states.h>
+#include <common/channel_type.h>
+#include <common/gossip_rcvd_filter.h>
+#include <common/gossip_store.h>
+#include <common/initial_channel.h>
+#include <common/lease_rates.h>
+#include <common/memleak.h>
+#include <common/peer_billboard.h>
+#include <common/peer_failed.h>
+#include <common/peer_io.h>
+#include <common/psbt_internal.h>
+#include <common/psbt_open.h>
+#include <common/read_peer_msg.h>
+#include <common/setup.h>
+#include <common/status.h>
+#include <common/subdaemon.h>
+#include <common/type_to_string.h>
+#include <common/wire_error.h>
+
+#define NUM_TX_MSGS (TX_RM_OUTPUT + 1)
```

Comment:
The idea is to NUM_TX_MSGS tied to the number of items in the enum. So if the enum gets changed INTERACTIVETX_NUM_TX_MSGS != NUM_TX_MSGS which will fail the assert on line 298. Sort of a defensive coding style to prevent future potential sneaky issues by making the failure explicit.